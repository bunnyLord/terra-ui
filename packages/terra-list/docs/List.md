# Terra List

 {insert description}

## Getting Started

- Install with [npmjs](https://www.npmjs.com): 
  - `npm install terra-list` 
  - `yarn install terra-list` 
- [Download the latest version](https://github.com/cerner/terra-list/archive/master.zip)
- Clone the repo: `git clone https://github.com/cerner/terra-list.git`
