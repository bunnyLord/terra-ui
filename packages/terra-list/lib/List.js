'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('../src/list.scss');

var _ListItem = require('./ListItem');

var _ListItem2 = _interopRequireDefault(_ListItem);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  className: _react.PropTypes.string,
  items: _react.PropTypes.arrayOf(_react.PropTypes.element),
  isDivided: _react.PropTypes.bool,
  hasChevrons: _react.PropTypes.bool
};

var defaultProps = {
  className: '',
  items: [],
  isDivided: false,
  hasChevrons: false
};

var List = function List(_ref) {
  var className = _ref.className,
      items = _ref.items,
      isDivided = _ref.isDivided,
      hasChevrons = _ref.hasChevrons,
      customProps = _objectWithoutProperties(_ref, ['className', 'items', 'isDivided', 'hasChevrons']);

  var listClassNames = (0, _classnames2.default)(['terra-List', { 'terra-List-divided': isDivided }, { 'terra-List-chevrons': hasChevrons }, className]);

  return _react2.default.createElement(
    'ul',
    _extends({}, customProps, { className: listClassNames }),
    items
  );
};

List.propTypes = propTypes;
List.defaultProps = defaultProps;
List.Item = _ListItem2.default;

exports.default = List;