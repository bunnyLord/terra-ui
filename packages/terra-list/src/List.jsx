import React, { PropTypes } from 'react';
import classNames from 'classnames';
import '../src/list.scss';
import ListItem from './ListItem';

const propTypes = {
  className: PropTypes.optionalString,
  items: PropTypes.arrayOf(PropTypes.element),
  isDivided: PropTypes.bool,
  hasChevrons: PropTypes.bool,
};

const defaultProps = {
  className: '',
  items: [],
  isDivided: false,
  hasChevrons: false,
};

const List = ({
    className,
    items,
    isDivided,
    hasChevrons,
    ...customProps
  }) => {
  const listClassNames = classNames([
    'terra-List',
    { 'terra-List-divided': isDivided },
    { 'terra-List-chevrons': hasChevrons },
    className,
  ]);

  return (
    <ul {...customProps} className={listClassNames}>
      {items}
    </ul>
  );
};

List.propTypes = propTypes;
List.defaultProps = defaultProps;
List.Item = ListItem;

export default List;
