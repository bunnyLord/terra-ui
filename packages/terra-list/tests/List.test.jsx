import React from 'react';
import initStoryshots from 'storyshots';
import List from '../src/List';

const defaultVariety = <List />;
const primaryVariety = <List name="primary" variant="terra-List--primary" />;

// Run snapshot tests for react-storybook
initStoryshots();

// Snapshot Tests
it('should render a default component', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
});

it('should render a primary component', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper).toMatchSnapshot();
});


// Prop Tests
it('should have the class terra-List--default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper.prop('className')).toContain('terra-List terra-List--default');
});

it('should have the class terra-List--primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper.prop('className')).toContain('terra-List terra-List--primary');
});


// Event Tests
it('should toggle the class u-selected on default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-List--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-List--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
});

it('should toggle the class u-selected on primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-List--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-List--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
