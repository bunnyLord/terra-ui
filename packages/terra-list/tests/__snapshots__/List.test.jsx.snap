exports[`test should render isDivided 1`] = `
<ul
  className="terra-List terra-List-divided">
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
</ul>
`;

exports[`test should render with items 1`] = `
<ul
  className="terra-List">
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
  <ListItem
    className=""
    hasChevron={false}
    isSelected={false} />
</ul>
`;
